### PR TITLE
Update data to 1.20.0 (Stable), 1.20.10 (Preview), and 1.20.20 (Preview)

### PR DESCRIPTION
**Pull request by outercloudstudio**
# TODO

- [x] Block Traits? ~Not sure how these work so not sure what needs to be updated~ I've read the docs and I think I get it now
- [x] `Introduced new entity filters "all_slots_empty" and "any_slot_empty"`
- [x] `Added spawn_item_event event to minecraft:spawn_entity component
- [x] `Removed "knockback_resistance" item component`
- [x] `Removed "dye_powder" item component`
- [x] `Released the minecraft:cooldown item component out of experimental`
- [x] `Released the minecraft:repairable item component out of experimental`
- [x] `Released the "minecraft:max_stack_size" item component out of experimental`
- [x] `Released the "minecraft:block_placer" item component out of experimental`
- [x] `Released the "minecraft:record" item component out of experimental`
- [x] `Released the "minecraft:shooter" item component out of experimental`
- [x] `Released the "minecraft:throwable" item component out of experimental`
- [x] `Released the "minecraft:projectile" item component out of experimental`
- [x] `Released the "minecraft:can_destroy_in_creative" item component out of experimental`
- [x] `Released the "minecraft:hover_text_color" item component out of experimental`
- [x] `Deprecate "minecraft:render_offsets" component in json formats 1.20.10 and higher`

# Concerns
- There are many duplicated files across version. I wonder if this could be reduced for better maintainability
- I'm not sure if bridge supports both preview and stable gametest versions as these change but bridge does not count them as separate. Ex: stable uses `1.3.0-beta` of the server package while preview only has `1.3.0`. My suggestion is if there are any conflicts, the version that minecraft stable uses should be used instead of the on that preview uses.
- ~Files are generally organized by version. I think it might be easier if this was not the case and we relied more on one file that contains multiple version checks.~ I think this might not work since I'm not sure how you would test the format version from inside a referenced file

### CODE REVIEW

**Comment by NguyenDuck** [comment]
Remove tracking the `dist` folder and `.lock` file, so thats will auto-generated, dont need tracked thats change

**Comment by outercloudstudio** [comment]
I'll add that to the gitnore

**Comment by NguyenDuck** [comment]
And deleted also 💯

**Comment by NguyenDuck** [comment]
Block Traits completed as #40

**Comment by ThomasOrs** [comment]
Block traits are brand new in experimental, the functionality can be seen here https://learn.microsoft.com/en-us/minecraft/creator/reference/content/blockreference/examples/blocktraits

**Comment by outercloudstudio** [comment]
Yep I'm polishing up NguyenDuck's commit for block traits right now.

**Comment by ThomasOrs** [comment]
To ensure everyone has the same dependencies, lock files should be committed to version control.
Adding the deno.lock file to gitignore is beyond the scope of this pull request.

**Comment by outercloudstudio** [comment]
There are some changes I need to complete before this is ready. Please hold off on the review.

**Comment by NguyenDuck** [comment]
* `minecraft:rotation` is deprecated in 1.19.80

**Comment by outercloudstudio** [comment]
> * `minecraft:rotation` is deprecated in 1.19.80

Can you link the change log that say this? I can't find it on the 1.19.80 release change log.

**Comment by NguyenDuck** [comment]
> Can you link the change log that say this? I can't find it on the 1.19.80 release change log.

The change log doesn't say that, but it's thrown error on used in minecraft

**Comment by outercloudstudio** [comment]
Alright thanks for catching that. I'll fix it now.

**Comment by ThomasOrs** [comment]
@outercloudstudio it was deprecated in favour of transformation
https://learn.microsoft.com/en-us/minecraft/creator/documents/update1.19.80
> [transformation](https://learn.microsoft.com/en-us/minecraft/creator/reference/content/blockreference/examples/blockcomponents/minecraftblock_transformation) - replaces the previous block rotation component (and adds scaling and translation)

https://learn.microsoft.com/en-us/minecraft/creator/reference/content/blockreference/examples/blockcomponents/minecraftblock_transformation

File: packages/minecraftBedrock/schema/block/v1.19.80/components/part_visibility.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
 	"additionalProperties": false,
 	"title": "Part Visibility",
 	"doNotSuggest": true,
-	"deprecationMessage": "Deprecated as of v1.19.80.20 - Sets conditions for when the block's different parts are visible."
+	"deprecationMessage": "Deprecated as of v1.19.80 - Sets conditions for when the block's different parts are visible."
```
Comment (ThomasOrs) [inline packages/minecraftBedrock/schema/block/v1.19.80/components/part_visibility.json:6]: This should point users to use `bone_visiblity`

File: packages/minecraftBedrock/schema/block/v1.19.80/components/rotation.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema",
+	"additionalProperties": false,
+	"title": "Rotation",
+	"doNotSuggest": true,
+	"deprecationMessage": "Deprecated as of v1.19.80 - The block's rotation around the center of the cube in degrees. The rotation is specified as [x, y, z]. Angles need to be in factors of 90."
```
Comment (ThomasOrs) [inline packages/minecraftBedrock/schema/block/v1.19.80/components/rotation.json]: This should point users to use `transformation`

**Review by ThomasOrs — CHANGES_REQUESTED** [review]

File: packages/minecraftBedrock/schema/block/v1.19.80/components/part_visibility.json (hunk 1 above)
Comment (ThomasOrs) [reply packages/minecraftBedrock/schema/block/v1.19.80/components/part_visibility.json]: Resolved in c7bc99d6f7eafb48496267af0aee1ef2362a84c8

**Review by ThomasOrs — APPROVED** [review]

**Pull request merged**